docs/01-tooling/02-better-minification-for-frontend-sources.md

```diff
@@ -1,6 +1,87 @@
 ## Better minification for the frontend sources
+
 ### Intro
+
+MediaWiki via [ResourceLoader][rl-min] uses [JavaScriptMinifier][] to minimize
+JavaScript files so that their size is as small as possible.
+
+Since the minification happens in the PHP server at runtime, tradeoffs were made
+so that the server could minify in a performant way, giving up size gains for
+speed.
```

Comment:
It might be worth mentioning that there's also a read through cache wrapping the ResourceLoader to minimise the impact of multiple requests because of the way that RL URLs are constructed (requesting multiple disjoint modules at once).

Reply:
Addressed in ebba045